src/main/java/frc/robot/subsystems/ArmControlSubsystem.java

```diff
@@ -0,0 +1,323 @@
+  private final PIDController mExtensionPID;
+  private final PIDController mPivotPID;
+
+  // This is called Ultra Instinct because setting this boolean to true removes the clamps
+  // so that if the belt skips, the operator can still run manually without the bad offsets
+  private boolean mUltraInstinct = false;
+
+  private double mPivotRelEncoderOffsetRot;
+  private double mCurrentPivotRotation;
+  private double mDesiredPivotRotation = ArmConstants.MIN_PIV_ANGLE_RAD;
+
+  private double mCurrentExtensionDistance = ArmConstants.ZERO_EXTENSION_IN;
+  private double mDesiredExtensionDistance = ArmConstants.MIN_EXT_LEN_IN;
+
+
+
+  private boolean mIsInitialized = false;
+
+  private final SendableChooser<ArmMotorMode> mChooser = new SendableChooser<>();
+
+
+  public ArmControlSubsystem() {
+
+    mPivotPID = new PIDController(1.4, 0, 0);
+    mPivotPID.setTolerance(ArmConstants.RESTING_PIV_TOLERANCE_RAD);
+
+    mExtensionPID = new PIDController(0.19, 0, 0);
+    mExtensionPID.setTolerance(ArmConstants.RESTING_EXT_TOLERANCE_IN);
+
+    mPivotRateLimiter = new SlewRateLimiter(ArmConstants.MAX_PIV_RATE_RAD_SEC);
+
+    mAbsPivEncoder.setConnectedFrequencyThreshold(ArmConstants.CONNECTION_THRESH_HZ);
+
+    mPivotRelEncoderOffsetRot =
+        -mAbsPivEncoder.getAbsolutePosition() * ArmConstants.PIVOT_ABS_ENC_TO_ROTATION
+            + ArmConstants.PIV_INIT_OFFSET_ROT;
+
+    setDefaultConfig();
+
+    SmartDashboard.putNumber("kG", ArmConstants.KG);
+
+    mChooser.addOption("Brake", ArmMotorMode.BRAKE);
+    mChooser.addOption("Coast", ArmMotorMode.COAST);
+    mChooser.setDefaultOption("Brake", ArmMotorMode.BRAKE);
+
+    SmartDashboard.putData("ArmMotorMode", mChooser);
+
+  }
+
+  public void setDefaultConfig() {
+    mRightPivotController.configFactoryDefault();
+    mLeftPivotController.configFactoryDefault();
+
+    mRightPivotController.follow(mLeftPivotController);
+
+    mRightPivotController.setInverted(TalonFXInvertType.OpposeMaster);
+    mLeftPivotController.setInverted(ArmConstants.LEFT_PIV_MOTOR_INVERTED);
+
+    updateModes();
+
+    mRightPivotController.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+    mLeftPivotController.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+
+    mLeftPivotController.setSelectedSensorPosition(0);
+    mRightPivotController.setSelectedSensorPosition(0);
+
+    mExtensionController.setInverted(false);
+
+    mExtensionEncoder.setPosition(0);
+
+    mExtensionController.burnFlash();
+  }
+
+  @Override
+  public void periodic() {
+    if (mChooser.getSelected() == ArmMotorMode.BRAKE && mIsInitialized) {
+      pivotPeriodic();
+      extensionPeriodic();
+    } else if (mChooser.getSelected() == ArmMotorMode.COAST) {
+      mDesiredPivotRotation = mCurrentPivotRotation;
+      mDesiredExtensionDistance = mCurrentExtensionDistance;
+    }
+
+    if (!mIsInitialized
+        && -mAbsPivEncoder.getAbsolutePosition() * ArmConstants.PIVOT_ABS_ENC_TO_ROTATION != 0
```

Comment:
The through bore encoder goes into an initialization phase before it begins to transmit sensor data, when it is at that phase it will read 0 regardless of the absolute position of the encoder. This is why we have this piece of logic here.

Reply:
Your reply doesn't directly address the review comment. The review has 2 points; an ask and a question. 
1. The ask is to fix the unnecessarily complex expression (-mAbsPivEncoder.getAbsolutePosition() * ArmConstants.PIVOT_ABS_ENC_TO_ROTATION != 0). 
- We don't write "if (-x * y != 0)" when we know for sure "y" is a non-zero constant. 
- In other word, we don't write "if (-x * 3 != 0)". Instead we write "if (x != 0)" to reduce compute cycles from roboRIO.

2. The questions is when mAbsPivEncoder.getAbsolutePosition() is zero and when it is very near zero. The question hints a potential problem, that is mAbsPivEncoder.getAbsolutePosition() is of floating point type (float, double).
Look up how floating point values are stored and you would see that people dont test for equality or its negation using "==" or "!=".
See https://stackoverflow.com/questions/1088216/whats-wrong-with-using-to-compare-floats-in-java

Reply:
done